privacyidea/lib/crypto.py

```diff
@@ -371,47 +326,52 @@ def decryptPassword(cryptPass, convert_unicode=False):
     # takes unicode strings!). But always returning unicode might break
     # other call sites of ``decryptPassword``. So we add the
     # keyword argument to avoid breaking compatibility.
-    from privacyidea.lib.utils import to_unicode
     hsm = get_hsm()
     try:
         ret = hsm.decrypt_password(cryptPass)
-    except Exception as exx:  # pragma: no cover
+    except Exception as exx:
         log.warning(exx)
         ret = FAILED_TO_DECRYPT_PASSWORD
     try:
         if convert_unicode:
-            ret = to_unicode(ret)
-    except Exception as exx:  # pragma: no cover
+            ret = _to_unicode(ret)
+    except Exception as exx:
         log.warning(exx)
         # just keep ``ret`` as a bytestring in that case
     return ret
 
 
 @log_with(log, log_exit=False)
 def decryptPin(cryptPin):
+    """
+
+    :param cryptPin: the encrypted pin
+    :type cryptPin: str, bytes, unicode
+    :return: the decrypted pin
+    :rtype: str
+    """
     hsm = get_hsm()
-    ret = hsm.decrypt_pin(cryptPin)
-    return ret
+    return _to_unicode(hsm.decrypt_pin(_to_bytes(cryptPin)))
 
 
 @log_with(log, log_entry=False)
 def encrypt(data, iv, id=0):
     '''
-    encrypt a variable from the given input with an initialiation vector
-
-    :param input: buffer, which contains the value
-    :type  input: buffer of bytes
-    :param iv:    initilaitation vector
-    :type  iv:    buffer (20 bytes random)
-    :param id:    contains the id of which key of the keyset should be used
-    :type  id:    int
-    :return:      encryted buffer
+    encrypt a variable from the given input with an initialisation vector
 
+    :param data: buffer, which contains the value
+    :type  data: bytes or str
+    :param iv:   initialisation vector
+    :type  iv:   bytes or str
+    :param id:   contains the key id of the keyset which should be used
+    :type  id:   int
+    :return:     encrypted and hexlified data
+    :rtype: str
 
     '''
     hsm = get_hsm()
-    ret = hsm.encrypt(data, iv, id)
-    return ret
```

Comment:
I used this step before the ``return`` since it was easier to debug. What do you tink?

Reply:
I think it is more 'pythonic' but You are right, debugging is easier :-)

Reply:
fixed in d709698